lib/core.js

```diff
@@ -176,6 +176,9 @@ declare var Math: {
     trunc(x: number): number;
 };
 
+type _NonNull<T, N: ?T> = T;
+type $NonNull<N> = _NonNull<*, N>;
```

Comment:
These magic types are needed for the addition. I'm not sure whats the best place to put them. Since they're only used here, I placed them here.